Correct websub signifiers

KaiTries opened the pull request:
This should solve issue #52 . 

The signifiers are now correctly located in the entity and the topic is now also correctly stated.

danaivach commented on yggdrasil-utils/src/main/java/org/hyperagents/yggdrasil/utils/impl/RepresentationFactoryHMASImpl.java:
```diff
@@ -94,418 +104,444 @@ private Signifier webSubSignifier(final WebSubSpecs specs,
-      .build();
-    final QualifiedValueSpecification registerArtifactInput = new QualifiedValueSpecification.Builder()
-      .addRequiredSemanticType(CORE.TERM.ARTIFACT.toString())
-      .setIRIAsString("http://example.org/artifact-rdf")
```
Consider replacing "http://example.org/artifact-rdf" with an IRI that uses the `baseUri`, e.g., `baseUri + "#artifact-rdf"`.

danaivach commented:
@KaiTries, all looks good regarding issue https://github.com/Interactions-HSG/yggdrasil/issues/52.

If additional changes are applied, e.g., regarding the config, please provide a brief overview here before merging. Thanks!

danaivach commented on README.md:
```diff
@@ -79,6 +79,228 @@ Run and build the Yggdrasil image in the project directory context
+The "known-artifacts" key takes a JsonArray as its value. Each Json Object contains the class and template for an artifact.
+The class is then used as a key to identify the artifacts at runtime. The template is simply the java class of the wanted artifact.
+When a virtual artifact is in the "known-artifacts" array users are then able to create instances of this class by using the makeArtifact endpoint
+in a workspace (needs to be a workspace that has an underlying cartago workspace). Correctly specified artifacts can then be used to remotely
```
cartago -> CArtAgO

danaivach commented on README.md:
```diff
@@ -79,6 +79,228 @@ Run and build the Yggdrasil image in the project directory context
+If the environment is disabled the "representation" parameter is also mandatory.
+
+##### Environment disabled
+If the environment is disabled, we do not use Cartago and therefore provide less functionality. This also means
```
cartago -> CArtAgO

danaivach commented on README.md:
```diff
@@ -79,6 +79,228 @@ Run and build the Yggdrasil image in the project directory context
+#### Environment enabled
+If the environment is enabled, the given yggdrasil instance will provide more funcionality in interacting with the environment.
+This is reflected in the more extendable configuration file. It is still possible to specifiy a workspace with a name and representation.
+If you do it this way NO underlying cartago workspace will be created!
```
cartago -> CArtAgO

danaivach commented on README.md:
```diff
@@ -79,6 +79,228 @@ Run and build the Yggdrasil image in the project directory context
+If you do it this way NO underlying cartago workspace will be created!
+
+##### workspaces
+The following configurations will all create an underlying cartago workspace instance.
```
cartago -> CArtAgO

danaivach commented on README.md:
```diff
@@ -79,6 +79,228 @@ Run and build the Yggdrasil image in the project directory context
+An agent consists of a name which will be given to the body artifact, an agent-uri which should point to the WebID of the agent and a callback-uri
+which should point to an endpoint of the agent that is capable of handling the update messages sent from yggdrasil.
+
+Specifying an agent in a workspace will automatically create a body artifact in said workspace as well as join the agent in the cartago workspace.
```
cartago -> CArtAgO

danaivach replied:
etc.

KaiTries commented:
implemented the comments by @danaivach . 

Since this branch built untop of my merge branch, there are some additional features that are part of this pull request:

- Complete reformatting such that the entire code base once again conforms to the checkstyle guidelines, refactored the code acording to pmd and checkBugs. 
- Renamed RdfStoreVerticle function "UpdateEntity" -> "ReplaceEntity" so that it is closer to its actual functionality
- Added new RdfStoreVerticle function "UpdateEntity" that actually updates a representation and does not replace already existing data. 
    - HttpHandler now utilizes this new function and when creating an entity through turtle, first a base representation is created and then the user sent representation is updated on top of that entity.
- Base Refactor of the configuration files for issue #56 
    - This came with some refactoring of the initialization code and the environment parser but does not cause any change in functionality
    - Added support for the "metadata" tag so that cartago entities can be initialized with additional metadata.
- Updated README with an overview of the config file structure

KaiTries commented:
still TODO:

- update test ttl files again

KaiTries commented:
Updated all test files and reformatted changes to be checkstyle conforming. 

Should be ready to merge

andreiciortea reviewed (approved):
Thanks @KaiTries!
